### PR TITLE
chore: Refactor artwork screen components

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **chore**

### Description

This PR refactors the artwork screen components (class component -> functional component).

Included components:

- [Artwork](http://www.google.com/search?q=src%2Flib%2FScenes%2FArtwork%2FArtwork.tsx&oq=src%2Flib%2FScenes%2FArtwork%2FArtwork.tsx&aqs=chrome.0.69i59j69i58.265j0j7&sourceid=chrome&ie=UTF-8)
- [AboutArtist](https://github.com/artsy/eigen/pull/5963/files#diff-965f51ed64081f5c91ee8412a7d39b312eeda42cf20c39c782846ce8dbdca0e9R11)
- [AboutWork](https://github.com/artsy/eigen/pull/5963/files#diff-c4350dd53d2574b98171d2cc567cda7dbee35de4a1cc2403655ba9426796933aR29)
- [AboutArtistHistory](https://github.com/artsy/eigen/pull/5963/files#diff-0b3cd4b942e8e851fbaeda8f7dd339e076f975ca5111db128b4cd40611b8e669R10)
- [CommercialInformation](https://github.com/artsy/eigen/pull/5963/files#diff-8a334cdddd01d8e1ed0beadbb9d9cfc7140539e4d15f2d25fa1598921a977bc1R9)
- [OtherWorks](https://github.com/artsy/eigen/pull/5963/files#diff-a9c0e57479f0875513869983f0062286c96761a292c4053ce9278d48d1de5452R14)

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Refactor and modernize artwork screen component - ole

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 97863ac2fe513bf5b0d2531257e7590188d660fa
</p>

File: src/lib/Scenes/Artwork/Artwork.tsx [hunk 1]
```diff
@@ -38,52 +39,51 @@ import { ContextCardFragmentContainer as ContextCard } from "./Components/Contex
...
-@screenTrack<Props>((props) => ({
-  context_screen: Schema.PageNames.ArtworkPage,
-  context_screen_owner_type: Schema.OwnerEntityTypes.Artwork,
-  context_screen_owner_slug: props.artworkAboveTheFold.slug,
-  context_screen_owner_id: props.artworkAboveTheFold.internalID,
-  availability: props.artworkAboveTheFold.availability,
-  acquireable: props.artworkAboveTheFold.is_acquireable,
-  inquireable: props.artworkAboveTheFold.is_inquireable,
-  offerable: props.artworkAboveTheFold.is_offerable,
-  biddable: props.artworkAboveTheFold.is_biddable,
```
Comment (olerichter00) [inline src/lib/Scenes/Artwork/Artwork.tsx:64]: Todo: Check if we really don't need these fields.

File: src/lib/Scenes/Artwork/Artwork.tests.tsx [hunk 2]
```diff
@@ -353,7 +353,7 @@ describe("Artwork", () => {
         },
         variables: {
           input: {
-            artwork_id: "completely-different-slug",
+            artwork_id: "my-special-artwork",
```
Comment (olerichter00) [inline src/lib/Scenes/Artwork/Artwork.tests.tsx:356]: This test checked if the `ArtworkMarkAsRecentlyViewedQuery` was called with the artwork ID of the prefetched artwork. However, the artwork ID should never change when we refetch an artwork (by ID). Therefore, we can safely update this spec.

File: src/lib/Scenes/Artwork/Components/AboutArtist.tsx [hunk 3]
```diff
@@ -11,55 +11,50 @@ interface AboutArtistProps {
...
+  )
 }
 
 export const AboutArtistFragmentContainer = createFragmentContainer(AboutArtist, {
   artwork: graphql`
     fragment AboutArtist_artwork on Artwork {
       artists {
         id
-        biography_blurb: biographyBlurb {
+        biographyBlurb {
```
Comment (MounirDhahri) [inline src/lib/Scenes/Artwork/Components/AboutArtist.tsx:57]: **praise:** thanks for the fix!

File: src/lib/Scenes/Artwork/Components/CommercialInformation.tsx [hunk 4]
```diff
@@ -32,52 +32,48 @@ interface CommercialInformationProps {
...
-        isPreview,
-        // @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
-        isClosed,
-        // @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
-        liveStartAt: liveStartsAt,
-        // @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
-        startAt: startsAt,
-        // @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
-        endAt: endsAt,
-      } = this.props.artwork.sale
```
Comment (MounirDhahri) [inline src/lib/Scenes/Artwork/Components/CommercialInformation.tsx:56]: **praise:** 👋 bye

**Review by MounirDhahri — APPROVED** [review]
This is a very satisfying PR to review 😄 Thanks

**Pull request merged**